src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -562,31 +581,27 @@ protected void Accepting(object message)
                             x => x.GracefulStop(settings.FlushWait, EndpointWriter.FlushAndStop.Instance))).ContinueWith(
                                 result =>
                                 {
-                                    if (result.IsFaulted)
-                                    {
-                                        if(result.Exception != null)
-                                            result.Exception.Handle(e => true);
-                                        return false;
-                                    }
-                                    return result.Result.All(x => x);
-                                }, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.AttachedToParent);
-
-                    var flushStatus = Task.WhenAll(_transportMapping.Values.Select(x => x.Shutdown())).ContinueWith(
-                                result =>
-                                {
-                                    if (result.IsFaulted)
+                                    if (result.IsFaulted || result.IsCanceled)
                                     {
                                         if (result.Exception != null)
                                             result.Exception.Handle(e => true);
                                         return false;
                                     }
                                     return result.Result.All(x => x);
-                                }, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.AttachedToParent);
+                                }, TaskContinuationOptions.ExecuteSynchronously);
+
+                    shutdownStatus.ContinueWith(tr => Task.WhenAll(_transportMapping.Values.Select(x => x.Shutdown())).ContinueWith(
+                              result =>
+                              {
+                                  if (result.IsFaulted || result.IsCanceled)
+                                  {
+                                      if (result.Exception != null)
+                                          result.Exception.Handle(e => true);
+                                      return false;
+                                  }
+                                  return result.Result.All(x => x) && tr.Result;
```

Comment:
what is the `x => x` thing doing?
Seems odd too compare each element to itself (?)